src/ts/proxyfowarder.ts

```diff
@@ -0,0 +1,159 @@
+import { keccak256, defaultAbiCoder, solidityKeccak256 } from "ethers/utils";
+import { Wallet } from "ethers/wallet";
+import { ReplayProtectionAuthority } from "./replayprotectionauthority";
+import { ChainID, ProxyAccountDeployer, ProxyAccountFactory } from "..";
+import { ForwardParams, Forwarder, ProxyCallData } from "./forwarder";
+
+/**
+ * A single library for approving meta-transactions and its associated
+ * replay protection. All meta-transactions are sent via proxy contracts.
+ */
+export class ProxyForwarder extends Forwarder<ProxyCallData> {
+  /**
+   * All meta-transactions are sent via an proxy contract.
+   * @param chainID Chain ID
+   * @param proxyHub Address of contract
+   * @param signer Signer's wallet
+   * @param replayProtectionAuthority Extends implementation ReplayProtectionAuthority
+   */
+  constructor(
+    chainID: ChainID,
+    proxyHub: ProxyAccountDeployer,
```

Comment:
Instead of the full ProxyAccountdeployer, it's probably better just to pass the address in here. That way people calling the constructor arent forced to create the Contract if they only have the address. You can create the deployer internally.

Also, all references to `proxyHub` should be renamed to `proxyAccountDeployer` or just `deployer` if the context is correct.

Reply:
mmm that means i'll need to check internally if its already been created? Ideally people should be using the factory anyway, so they wont need to create it.

Reply:
You wont have to check if it's already been created, you'll just need to attach to supplied address when you need to, currently this only in one function:
```
 public async createProxyContract() {
    const deployed = await this.proxyHub
      .connect(this.signer)
      .accounts(this.signer.address);
```
which you could become:

```
 public async createProxyContract() {
   const proxyAccountFactory = new ProxyAccountFactory(this.signer).attach(this.proxyAccountHubAddress)
    const deployed = await proxyAccountFactory
      .connect(this.signer)
      .accounts(this.signer.address);
```

Reply:
I've changed this around so there is a isProxyContractDeployed() function. I'll add that code in with a try/catch.